Auto sync upstream 2022 03 17

nunnatsa opened the pull request:
<!-- Thanks for sending a pull request! -->

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release notes**:
<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access)
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`.
-->
```release-note
Auto sync upstream 2022 03 17
```

nirarg commented:
/test e2e-kubevirt

nirarg commented:
/test e2e-kubevirt

nirarg commented:
/test e2e-kubevirt

nirarg commented:
/test e2e-kubevirt

nirarg commented:
/test e2e-kubevirt

bardielle commented:
/test e2e-kubevirt

bardielle commented:
/retest

bardielle commented on pkg/actuator/actuator.go:
```diff
@@ -39,22 +40,31 @@ const (
+		return err
+	}
+
+	klog.Errorf("%v error: %v", machineScope.GetMachineName(), err)
```
change the error msg "error creating machineScopre

bardielle commented on cmd/manager/main.go:
```diff
@@ -141,8 +141,11 @@ func main() {
 	kubevirtVM := kubevirt.New(infraClusterClient)
 
 	// Initialize machine actuator.
```
I think you need to change to commit msg: TenantclusterClient: use configMap instead of ....

bardielle commented on pkg/testutils/stubs.go:
```diff
@@ -0,0 +1,245 @@
+	accessMode       = corev1.ReadWriteOnce
+	memory           = "123456M"
+	storage          = "666Gi"
+	CPU              = 77
```
cpu or move it to be in line 20

nirarg replied:
Done

bardielle commented on pkg/testutils/stubs.go:
```diff
@@ -0,0 +1,245 @@
+	CPU              = 77
+	storageClassName = "test-storage-class"
+	SrcUserData      = "{\"ignition\":{\"config\":{\"merge\":[{\"source\":\"https://192.168.123.15:22623/config/worker\"}]},\"security\":{\"tls\":{\"certificateAuthorities\":[{\"source\":\"data:text/plain;charset=utf-8;base64,LS0tLCg==\"}]}},\"version\":\"3.1.0\"}}"
+	FullUserDataFmt  = "{\"ignition\":{\"config\":{\"merge\":[{\"source\":\"https://192.168.123.15:22623/config/worker\"}]},\"security\":{\"tls\":{\"certificateAuthorities\":[{\"source\":\"data:text/plain;charset=utf-8;base64,LS0tLCg==\"}]}},\"version\":\"3.1.0\"},\"storage\":{\"files\":[{\"contents\":{\"source\":\"data:,%s\"},\"filesystem\":\"root\",\"mode\":420,\"path\":\"/etc/hostname\"}]}}"
```
FullUserDataFmt ->fullUserDataFmt or move it to be in line 20

nirarg replied:
Done

bardielle commented on pkg/testutils/stubs.go:
```diff
@@ -0,0 +1,245 @@
+	storage          = "666Gi"
+	CPU              = 77
+	storageClassName = "test-storage-class"
+	SrcUserData      = "{\"ignition\":{\"config\":{\"merge\":[{\"source\":\"https://192.168.123.15:22623/config/worker\"}]},\"security\":{\"tls\":{\"certificateAuthorities\":[{\"source\":\"data:text/plain;charset=utf-8;base64,LS0tLCg==\"}]}},\"version\":\"3.1.0\"}}"
```
SrcUserData -> srcUserData or move it to be in line 20

nirarg replied:
Done

bardielle commented on pkg/kubevirt/kubevirt.go:
```diff
@@ -51,10 +51,7 @@ func (m *manager) Create(machineScope machinescope.MachineScope, userData []byte
-	if err != nil {
-		return err
-	}
+	secretFromMachine := machineScope.CreateIgnitionSecretFromMachine(fullUserData)
```
Change the commit msg to be more informative - Using references, renaming methods etc...

bardielle commented on pkg/machinescope/machine_scope.go:
```diff
@@ -438,46 +401,48 @@ func (s *machineScope) setMachineAnnotationsAndLabels(vm *kubevirtapiv1.VirtualM
+	klog.Infof("%s - syncNetworkAddresses: successfully synced machine.Status.Addresses to %s", s.GetMachineName(), networkAddresses)
 }
+
+// TODO: update the phase of the machine
```
remove those line from comment and add a ticket ....

nirarg commented on pkg/actuator/actuator.go:
```diff
@@ -63,53 +73,73 @@ func (a *Actuator) handleMachineError(machine *machinev1.Machine, err error, eve
-		fmtErr := fmt.Errorf(vmsFailFmt, vm.GetMachineName(machine), createEventAction, err)
+	if err := a.providerVM.Create(machineScope); err != nil {
+		fmtErr := fmt.Errorf(vmsFailFmt, machineScope.GetMachineName(), createEventAction, err)
 		return a.handleMachineError(machine, fmtErr, createEventAction)
```
Removed

Regarding all comments in actuator:
I didn't do the refactoring to this file yet, just copied code from/to it
I will add additional commit for refactoring it, including all relevant comments

nirarg commented on pkg/machinescope/machine_scope.go:
```diff
@@ -329,10 +303,55 @@ func (s *machineScope) buildVMITemplate(namespace string) (*kubevirtapiv1.Virtua
 	return fmt.Sprintf("%s-ignition", virtualMachineName)
 }
 
+func (s *machineScope) CreateIgnitionSecretFromMachine() (*corev1.Secret, error) {
```
1. userData provided as parameter
2. Remove returned error (always nil)

nirarg commented on pkg/kubevirt/utils.go:
```diff
@@ -1,108 +0,0 @@
-		clusterID, ok = machine.Labels[upstreamMachineClusterIDLabel]
-	}
-	return clusterID, ok
-}
```
That file was deleted and `addHostnameToUserData` was moved to kubevirt.go

nirarg commented:
/retest

bardielle commented on pkg/machinescope/machine_scope_test.go:
```diff
@@ -1,27 +1,405 @@
+	}
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			machine, err := testutils.StubMachine()
```
I think that this part of the function is too long 
If someone will add a part to the sync function it's too complicated to understand where should the test be added

bardielle commented on pkg/machinescope/machine_scope_test.go:
```diff
@@ -1,27 +1,405 @@
+				t.Fatalf("Error durring machineScope creation: %v", err)
+			}
+
+			vmName := "test-vm-name"
+			vmNamespace := "test-vm-namespace"
+			vmID := "test-vm-id"
+			machineType := "test-machine-type"
+
+			vm := kubevirtapiv1.VirtualMachine{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      vmName,
+					Namespace: vmNamespace,
+					UID:       types.UID(vmID),
+				},
+			}
```
why this part is not in testutils.StubVirtualMachine?

bardielle commented on pkg/machinescope/machine_scope_test.go:
```diff
@@ -1,27 +1,405 @@
-func TestAddHostNameToUserData(t *testing.T) {
-	userData := "{\"ignition\":{\"config\":{\"merge\":[{\"source\":\"https://192.168.123.15:22623/config/worker\"}]},\"security\":{\"tls\":{\"certificateAuthorities\":[{\"source\":\"data:text/plain;charset=utf-8;base64,LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSURFRENDQWZpZ0F3SUJBZ0lJS01BN3hKRW5lWXd3RFFZSktvWklodmNOQVFFTEJRQXdKakVTTUJBR0ExVUUKQ3hNSmIzQmxibk5vYVdaME1SQXdEZ1lEVlFRREV3ZHliMjkwTFdOaE1CNFh ...
+func TestGetMachine(t *testing.T) {
+	machine, err := testutils.StubMachine()
+	if err != nil {
+		t.Fatalf("Error durring stubMachine creation: %v", err)
+	}
+	machineScope, err := New().CreateMachineScope(machine, testutils.InfraNamespace, testutils.InfraID)
+	if err != nil {
+		t.Fatalf("Error durring machineScope creation: %v", err)
+	}
```
It seems that most of the test scenrio start with those lines
I think you should move it to "BeforeEachTest" or initializeTest function

bardielle commented on cmd/manager/main.go:
```diff
@@ -88,6 +87,8 @@ func main() {
 	logf.SetLogger(logf.ZapLogger(false))
 	entryLog := log.WithName("entrypoint")
 
+	entryLog.Info("start kubevirt machine controller")
```
change the commit msg of refactor main to be more specific

bardielle commented on pkg/actuator/actuator.go:
```diff
@@ -21,6 +21,7 @@ import (
 	"fmt"
 
 	machinev1 "github.com/openshift/machine-api-operator/pkg/apis/machine/v1beta1"
+	"github.com/pkg/errors"
```
change the commit msg of refactor actuator to be more specific

bardielle replied:
OR squash it with the first commit of actutor...

bardielle commented on pkg/machinescope/machine_scope_test.go:
```diff
@@ -1,27 +1,424 @@
+		},
+	}
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
```
:+1: 
The function looks good

bardielle reviewed (approved):
Great job!

bardielle commented:
/lgtm

nirarg commented:
/test e2e-kubevirt

nirarg commented:
/test e2e-kubevirt

nirarg commented:
/test e2e-kubevirt

nirarg commented:
/test e2e-kubevirt

nirarg committed:
Change OWNERS file

nunnatsa committed:
Automatic sync with upstream

This PR adds a new github action to daily sync with upstream. It checks if there are new changes in upstream, and if there are, it creates a PR to sync.

The new PR is created by hard reseting to the upstream main branch, then restoring the downstream modified files.

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

nunnatsa committed:
add script to sync with upstream

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

nunnatsa committed:
create PR

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

0xFelix committed:
Fix typo in kubevirtci::fetch_kubevirtci

KUEVIRTCI_TAG --> KUBEVIRTCI_TAG

Signed-off-by: Felix Matouschek <fmatouschek@redhat.com>

k8s-ci-robot committed:
Merge pull request #112 from 0xFelix/patch-1

Fix typo in kubevirtci::fetch_kubevirtci

davidvossel committed:
ensure we never reconcile an externally managed kvcluster

Signed-off-by: David Vossel <davidvossel@gmail.com>

davidvossel committed:
Add e2e test to ensure finalizer isn't added for external clusters

Signed-off-by: David Vossel <davidvossel@gmail.com>

k8s-ci-robot committed:
Merge pull request #116 from davidvossel/fix-external-managed

Fix external managed kubevirt clusters don't get a finalizer

nirarg commented:
/lgtm
